zerver/lib/users.py

```diff
@@ -147,12 +150,20 @@ def validate_and_construct_bot_email(short_name_raw: str, realm: Realm) -> str:
     try:
         email = Address(username=short_name, domain=realm.get_bot_domain()).addr_spec
     except InvalidFakeEmailDomainError:
-        raise JsonableError(
-            _(
-                "Can't create bots until FAKE_EMAIL_DOMAIN is correctly configured.\n"
-                "Please contact your server administrator."
+        if for_editing:
```

Comment:
I am not sure this is good. I cannot think of a common string that we can use for both cases right now. Maybe `InvalidFakeEmailDomainError` can be handled in the caller and not here.